api.ts

```diff
@@ -1044,29 +1051,30 @@ export interface SimpleAutocompleteOption {
  * }
  * ```
  */
-export function simpleAutocomplete(
+export function simpleAutocomplete<T extends ParameterType.Number | ParameterType.String>(
   search: string | undefined,
-  options: Array<string | number | SimpleAutocompleteOption>,
+  options: Array<TypeMap[ParameterTypeMap[T]] | SimpleAutocompleteOption<T>>,
 ): Promise<MetadataFormulaObjectResultType[]> {
   const normalizedSearch = (search || '').toLowerCase();
   const filtered = options.filter(option => {
     const display = typeof option === 'string' || typeof option === 'number' ? option : option.display;
     return display.toString().toLowerCase().includes(normalizedSearch);
   });
-  const metadataResults = filtered.map(option => {
+  const metadataResults: MetadataFormulaObjectResultType[] = [];
+  for (const option of filtered) {
     if (typeof option === 'string') {
-      return {
+      metadataResults.push({
         value: option,
         display: option,
-      };
+      });
     } else if (typeof option === 'number') {
-      return {
+      metadataResults.push({
         value: option,
         display: option.toString(),
-      };
+      });
     }
-    return option;
```

Comment:
Does this silently drop options that are strings or numbers? Should we error on that instead?

Reply:
let me do that.